screen/wallets/entropyGenerator.js

```diff
@@ -0,0 +1,438 @@
+  const labelProps = useRoute().params.label || '';
+
+  const onDrawing = (mode, g) => {
+    const newPaths = [...paths];
+    // Modes follow syntax of path on SVG: read more here https://css-tricks.com/svg-path-syntax-illustrated-guide/
+    newPaths[paths.length - 1].segments.push(`${mode} ${g.x} ${g.y}`);
+    setPaths(newPaths);
+  };
+
+  const pan = Gesture.Pan()
+    .onStart((g) => {
+      onDrawing('M', g);
+    })
+    .onUpdate((g) => {
+      onDrawing('L', g);
+    })
+    .onEnd(() => {
+      setPaths([...paths, { segments: [], color }]);
+    })
+    .minDistance(1);
+
+  const onClearDrawingButtonClick = () => {
+    setPaths([{ segments: [], color }]);
+  };
+
+  const { colors } = useTheme();
+  const { addWallet, saveToDisk, isAdancedModeEnabled, wallets, isTestModeEnabled } = useContext(BlueStorageContext);
+  const [isLoading, setIsLoading] = useState(true);
+  const [walletBaseURI, setWalletBaseURI] = useState('');
+  const [selectedIndex, setSelectedIndex] = useState(0);
+  const [label, setLabel] = useState(labelProps || '');
+  const [isAdvancedOptionsEnabled, setIsAdvancedOptionsEnabled] = useState(false);
+  const [isTestMode, setIsTestMode] = useState(false);
+  const [selectedWalletType, setSelectedWalletType] = useState(selectedWalletTypeProps || false);
+  const [backdoorPressed, setBackdoorPressed] = useState(1);
+
+  const [entropy, setEntropy] = useState();
+  const [entropyButtonText, setEntropyButtonText] = useState(loc.wallets.add_entropy_provide);
+  const stylesHook = {
+    advancedText: {
+      color: colors.feeText,
+    },
+    label: {
+      borderColor: colors.formBorder,
+      borderBottomColor: colors.formBorder,
+      backgroundColor: colors.inputBackgroundColor,
+    },
+    noPadding: {
+      backgroundColor: colors.elevated,
+    },
+    root: {
+      backgroundColor: colors.elevated,
+      marginBottom: 30,
+    },
+    lndUri: {
+      borderColor: colors.formBorder,
+      borderBottomColor: colors.formBorder,
+      backgroundColor: colors.inputBackgroundColor,
+    },
+  };
+
+  useEffect(() => {
+    AsyncStorage.getItem(isTestMode ? AppStorage.TEST_LNDHUB : AppStorage.LNDHUB)
+      .then((url) => setWalletBaseURI(url || (isTestMode ? 'https://tnlndhub.mintlayer.org' : 'https://lndhub.io')))
+      .catch(() => setWalletBaseURI(''));
+    isTestModeEnabled().then(setIsTestMode);
+    isAdancedModeEnabled()
+      .then(setIsAdvancedOptionsEnabled)
+      .finally(() => setIsLoading(false));
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [isAdvancedOptionsEnabled]);
+
+  const entropyGenerated = (newEntropy) => {
+    let entropyTitle;
+    if (!newEntropy) {
+      entropyTitle = loc.wallets.add_entropy_provide;
+    } else if (newEntropy.length < 32) {
+      entropyTitle = loc.formatString(loc.wallets.add_entropy_remain, {
+        gen: newEntropy.length,
+        rem: 32 - newEntropy.length,
```

Comment:
What happens with this remainder? If I have 31 bytes what happens?

Reply:
@muursh  removing function , it's not is used